test/blackbox-tests/test-cases/github3046/run.t

```diff
@@ -17,18 +17,12 @@ are given as paramters
 
   $ dune init lib foo --ppx="foo bar"
   dune: option `--ppx': invalid element in list (`foo bar'): expected a valid
-        dune atom
+        dune 'ppx' name
   Usage: dune init [OPTION]... INIT_KIND NAME [PATH]
   Try `dune init --help' or `dune --help' for more information.
   [1]
 
 `dune init lib foo --public="some/invalid&name!"` returns an informative parsing error
 
   $ dune init lib foo --public="some/invalid&name!"
-  dune: option `--public': invalid component name `some/invalid&name!'
-        Library names must be non-empty and composed only of the
-        following
-        characters: 'A'..'Z', 'a'..'z', '_' or '0'..'9'.
-  Usage: dune init [OPTION]... INIT_KIND NAME [PATH]
-  Try `dune init --help' or `dune --help' for more information.
-  [1]
+  Success: initialized library component named foo
```

Comment:
Should be sucess because --public name of a library doesn't have a validation logic yet.